tests/spread/dependencies/strict-dependencies/charm/requirements.txt

```diff
@@ -0,0 +1 @@
+ops==2.5.1
```

Comment:
so just to clarify, this requirements file has ops at 2.5.1, but the charm-binary-python-packages property is ops>=2.6, and that's accepted? How come?

Reply:
I see from the task.yaml comments that the version installed is 2.5.1, but isn't that very surprising to users that explicitly requested >2.6?

Reply:
Hmmm good point. I didn't consider them "requesting" ops >= 2.6 here, but rather allowing it as binary. However, for someone used to the old dependency management the principle of least surprise says I should probably enforce than strict dependencies only allow a package name in this field.

Reply:
Strict mode now only allows package names in `charm-binary-python-packages`